Forms/FormBuilder.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace SilverStripe\Forms;
+
+use SilverStripe\Control\Controller;
+use SilverStripe\ORM\DataObject;
+
+/**
+ * A service which can generate a form for a given record and controller
+ */
+interface FormBuilder {
+
+	/**
+	 * @return DataObject
+	 */
+	public function getRecord();
```

Comment:
This method is necessary for extensions on controllers; If generating fields for a form, extensions on the current controller will need to know (in many cases) which record is being scaffolded for. Note that this occurs in many places before the Form itself has been constructed. It's not intended that the builder generates the record itself, only that it holds it.